src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -2669,24 +2669,23 @@ self =>
       val lhs = commaSeparated(stripParens(noSeq.pattern2()))
       val tp = typedOpt()
       val rhs =
-        if (tp.isEmpty || in.token == EQUALS) {
-          accept(EQUALS)
-          if (!tp.isEmpty && newmods.isMutable &&
-              (lhs.toList forall (_.isInstanceOf[Ident])) && in.token == USCORE) {
-            in.nextToken()
-            tp match {
-              case SingletonTypeTree(Literal(Constant(_))) =>
-                syntaxError(tp.pos, "default initialization prohibited for literal-typed vars", skipIt = false)
-              case _ =>
-            }
-            newmods = newmods | Flags.DEFAULTINIT
-            EmptyTree
-          } else {
-            expr()
-          }
-        } else {
+        if (!tp.isEmpty && in.token != EQUALS) {
           newmods = newmods | Flags.DEFERRED
           EmptyTree
+        } else {
+          accept(EQUALS)
+          expr() match {
+            case x if !tp.isEmpty && newmods.isMutable && lhs.toList.forall(_.isInstanceOf[Ident]) && isWildcard(x) =>
```

Comment:
```suggestion
            case x if !tp.isEmpty && newmods.isMutable && lhs.forall(_.isInstanceOf[Ident]) && isWildcard(x) =>
```
`lhs` already is a list.

Reply:
Oh I didn't see your comment. I noticed the toList and didn't follow up. Not everything in the compiler API is a list or iterator; I suspect earlier code lacked forall et al. There ought to be -Xlint:monadic or somesuch to detect redundant calls.

Reply:
I used "commit suggestion" for the first time...